services/datamanager/builtin/builtin_test.go

```diff
@@ -635,6 +658,52 @@ func TestSyncDisabled(t *testing.T) {
 	test.That(t, len(mockService.getUploadedFiles()), test.ShouldEqual, 3)
 }
 
+func TestSessionIDAfterUpdate(t *testing.T) {
+	rpcServer, _ := buildAndStartLocalServer(t)
+	defer func() {
+		err := rpcServer.Stop()
+		test.That(t, err, test.ShouldBeNil)
+	}()
+
+	testCfg := setupConfig(t, configPath)
+	dmCfg, err := getDataManagerConfig(testCfg)
+	test.That(t, err, test.ShouldBeNil)
+	dmCfg.SyncIntervalMins = syncIntervalMins
+
+	// Make the captureDir where we're logging data for our arm.
+	captureDir := "/tmp/capture"
+	armDir := captureDir + "/arm/arm1/"
+	defer resetFolder(t, armDir)
+
+	// Initialize the data manager and update it with our config.
+	dmsvc := newTestDataManager(t, "arm1", "")
+	dmsvc.SetSyncerConstructor(getTestSyncerConstructor(t, rpcServer))
+	err = dmsvc.Update(context.TODO(), testCfg)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Verify SessionID is the same after reconfiguration of something in the config,
+	// but the data manager service config is the same
+	sessionID0 := dmsvc.GetSessionID()
+	dmCfg.SyncIntervalMins += 2
+	err = dmsvc.Update(context.TODO(), testCfg)
+	test.That(t, err, test.ShouldBeNil)
+	sessionID1 := dmsvc.GetSessionID()
+	test.That(t, sessionID0, test.ShouldEqual, sessionID1)
+
+	// Get and change capture frequency
+	mp := testCfg.Components[0].ServiceConfig[0].Attributes["capture_methods"].([]interface{})[0]
+	mp.(map[string]interface{})["capture_frequency_hz"] = 200
+
+	// Check the session is different after data manager service configuration has changed.
```

Comment:
This tests that the session is different by the time you call `dmsvc.GetSessionID()`, but I believe that the collectors were initialized with the previous session ID. Explanation in another comment directly on the code